addon/services/session.js

```diff
@@ -212,16 +212,31 @@ export default Ember.Service.extend(Ember.Evented, {
     });
     ```
 
+    This function accepts any number of arguments but two at minimum. The first argument is
+    always the name of the used authorizer and the last argument is always a function (`block`) to
+    be called with the authorization data generated by the authorizer. Between those two arguments,
+    zero or more additional arguments can be given and they will be bassed on as is to the authorizer
+    when called.
+
     @method authorize
     @param {String} authorizer The authorizer to authorize the block with
     @param {Function} block The block of code to call with the authorization data generated by the authorizer
     @public
   */
-  authorize(authorizerFactory, block) {
+  authorize(...args) {
+    Ember.assert('session.authorize() must be called with at least two parameters', args.length >= 2);
+    Ember.assert('The first parameter to session.authorize() must be string with the name of the used authorizer', typeof args[0] === 'string');
+    Ember.assert('The last parameter to session.authorize() must be function', typeof args[args.length - 1] === 'function');
```

Comment:
After thinking about this longer I don't actually think this is the way to go. Passing additional arguments as individual arguments before the callback means that the calling site has to know the signature of the authorizer in use which is not actually the case for e.g. the `DataAdapterMixin` where the authorizer is configurable and it should just work with all authorizers out of the box. Instead what would be easier would be to add an `options` argument that combines all the additional arguments for the authorizer and that would be passed for all authorizers so that the calling site doesn't have to know the signature.

As the `options` argument would have to default to `{}` though it cannot be the second argument but would have to be the last one, making it a breaking change. A solution might be to also make the new API the opt in for promise based authorization (which would actually make sense anyway) so that when the second argument is a function it would still use the old API and if the second argument is an object that would be the opt-in for promise based authorization, e.g.: 

```js
this.get('session').authorize('authorizer:custom').then(…)
```

The problem is that jQuery's `beforeSend` (as used in the `DataAdapterMixin`) doesn't support promises so we'd need to find a solution for that then.